docs/pages/[id].js

```diff
@@ -0,0 +1,65 @@
+// @flow strict
+import { MDXRemote } from 'next-mdx-remote';
+import { serialize } from 'next-mdx-remote/serialize';
+import { type Node } from 'react';
+
+import { getDocByRoute, getAllMarkdownPosts } from '../utils/mdHelper.js';
+import MarkdownPage from '../components/MarkdownPage.js';
+
+type MDXRemoteSerializeResult = {|
+  compiledSource: string,
+
+  frontmatter?: { [key: string]: string },
+|};
+
+type Props = {|
+  content: MDXRemoteSerializeResult,
+  meta: {|
+    title: string,
+    badge: 'pilot' | 'deprecated',
+    component?: boolean,
+    description: string,
+  |},
+  pageSourceUrl: string,
+|};
+
+export default function DocumentPage({ content, meta, pageSourceUrl }: Props): Node {
+  return (
+    <MarkdownPage meta={meta} pageSourceUrl={pageSourceUrl}>
+      <MDXRemote {...content} />
+    </MarkdownPage>
+  );
+}
+
+export async function getStaticProps(context: {| params: {| id: string |} |}): Promise<{|
+  props: {| meta: { [key: string]: string }, content: {||}, pageSourceUrl: string |},
+|}> {
+  const { id } = context.params;
+  const { meta, content } = getDocByRoute(id);
+  const mdxSource = await serialize(content);
+
+  return {
+    props: {
+      meta,
+      content: mdxSource,
+      pageSourceUrl: `https://github.com/pinterest/gestalt/tree/master/docs/pages/markdown/${id}.md`,
+    },
+  };
+}
+
+export async function getStaticPaths(): Promise<{|
```

Comment:
The only paths valid paths that should exist outside of traditional pages are in the "markdown" folder

Reply:
For context 
https://nextjs.org/docs/basic-features/data-fetching/get-static-paths

> If a page has [Dynamic Routes](https://nextjs.org/docs/routing/dynamic-routes) and uses getStaticProps, it needs to define a list of paths to be statically generated.

> When you export a function called getStaticPaths (Static Site Generation) from a page that uses dynamic routes, Next.js will statically pre-render all the paths specified by getStaticPaths.

> getStaticPaths must be used with getStaticProps. You cannot use it with [getServerSideProps](https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props).